pkg/trimaran/lowriskovercommitment/lowriskovercommitment.go

```diff
@@ -0,0 +1,258 @@
+
+	pluginConfig "sigs.k8s.io/scheduler-plugins/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/trimaran"
+)
+
+const (
+	// Name : name of plugin
+	Name = "LowRiskOverCommitment"
+
+	// MaxVarianceAllowance : allowed value from the maximum variance (to avoid zero divisions)
+	MaxVarianceAllowance = 0.99
+)
+
+// LowRiskOverCommitment : scheduler plugin
+type LowRiskOverCommitment struct {
+	handle       framework.Handle
+	eventHandler *trimaran.PodAssignEventHandler
+	collector    *trimaran.Collector
+	args         *pluginConfig.LowRiskOverCommitmentArgs
+}
+
+// New : create an instance of a LowRiskOverCommitment plugin
+func New(obj runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	klog.V(4).InfoS("Creating new instance of the LowRiskOverCommitment plugin")
+	// cast object into plugin arguments object
+	args, ok := obj.(*pluginConfig.LowRiskOverCommitmentArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type LowRiskOverCommitmentArgs, got %T", obj)
+	}
+	collector, err := trimaran.NewCollector(&args.TrimaranSpec)
+	if err != nil {
+		return nil, err
+	}
+	klog.V(4).InfoS("Using LowRiskOverCommitmentArgs", "smoothingWindowSize", args.SmoothingWindowSize,
+		"riskLimitWeight", args.RiskLimitWeight)
+
+	podAssignEventHandler := trimaran.NewNoMemory()
+	handle.SharedInformerFactory().Core().V1().Pods().Informer().AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch t := obj.(type) {
+				case *v1.Pod:
+					return isAssigned(t)
+				case cache.DeletedFinalStateUnknown:
+					if pod, ok := t.Obj.(*v1.Pod); ok {
+						return isAssigned(pod)
+					}
+					utilruntime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Pod", obj))
+					return false
+				default:
+					utilruntime.HandleError(fmt.Errorf("unable to handle object: %T", obj))
+					return false
+				}
+			},
+			Handler: podAssignEventHandler,
+		},
+	)
+
+	pl := &LowRiskOverCommitment{
+		handle:       handle,
+		eventHandler: podAssignEventHandler,
+		collector:    collector,
+		args:         args,
+	}
+	return pl, nil
+}
+
+// Score : evaluate score for a node
+func (pl *LowRiskOverCommitment) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	klog.V(6).InfoS("Calculating score", "pod", klog.KObj(pod), "nodeName", nodeName)
+	score := framework.MinNodeScore
+	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
+	if err != nil {
+		return score, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
+	}
+	// get node metrics
+	metrics, _ := pl.collector.GetNodeMetrics(nodeName)
+	if metrics == nil {
+		klog.InfoS("Failed to get metrics for node; using minimum score", "nodeName", nodeName)
+		return score, nil
+	}
+	// calculate score
+	totalScore := pl.computeRank(metrics, nodeInfo.Node(), pod) * float64(framework.MaxNodeScore)
```

Comment:
Why don't you pass in NodeInfo, and then use nodeInfo.Pods to obtain the pods?

Reply:
Thanks, done.